% \documentclass[a4,center,fleqn]{NAR}
\documentclass{article}

%\VignetteEngine{knitr::knitr}
%\VignetteIndexEntry{draft}

% To change between an NAR manuscript and a format-less article,
% toggle both the above (\documentclass) and the below (\NARtrue for NAR)
\newif\ifNAR
% \NARtrue
\NARfalse

\ifNAR
  \usepackage{NAR-natbib}

  % Enter dates of publication
  \copyrightyear{2015}
  \pubdate{X}
  \pubyear{2015}
  \jvolume{X}
  \jissue{X}
\else
  \usepackage{fullpage}
\fi

%\usepackage{geometry}                % See geometry.pdf to learn the layout options. There are lots.
%\geometry{letterpaper}                   % ... or a4paper or a5paper or ...
%\geometry{landscape}                % Activate for rotated page geometry
%\usepackage[parfill]{parskip}    % Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{epstopdf}
\usepackage{hyperref}
\DeclareGraphicsRule{.tif}{png}{.png}{`convert #1 `dirname #1`/`basename #1 .tif`.png}

% JOHN ADDED
%\usepackage{lineno}
%\linenumbers

\begin{document}

\title{A Nested Parallel Experiment Demonstrates Differences in Intensity-Dependence Between RNA-Seq and Microarrays}
%  \author{
% David G. Robinson$^{1}$,
% Jean Wang$^{1}$, and John D. Storey$^{1,2,3 \; \dagger}$
% \\ \footnotesize 1. Lewis-Sigler Institute for Integrative Genomics, Princeton University, Princeton, NJ 08544
% \\ \footnotesize 2. Center for Statistics and Machine Learning, Princeton University, Princeton, NJ 08544
% \\ \footnotesize 3. Department of Molecular Biology, Princeton University, Princeton, NJ 08544
% \\ \footnotesize $^\dagger$ To whom correspondence should be addressed: {\tt jstorey@princeton.edu}
% }
\date{}                                           % Activate to display a given date or no date

\author{%
David G. Robinson$^{1}$,
Jean Y. Wang$^{1}$, and John D. Storey$^{1,2,3}$
\footnote{To whom correspondence should be addressed. Email: jstorey@princeton.edu}}

\ifNAR
  \address{%
  $^{1}$Lewis-Sigler Institute for Integrative Genomics, Princeton University, Carl Icahn Laboratory, Princeton, NJ 08544
  $^{2}$Center for Statistics and Machine Learning, Princeton University, Princeton, NJ 08544
  $^{3}$Department of Molecular Biology, Princeton University, Princeton, NJ 08544}
  \history{%
  }
\fi

\maketitle

<<options, echo = FALSE>>=
library(knitr)
opts_chunk$set(tidy = TRUE, message = FALSE, cache = TRUE,
               echo = FALSE, results = "hide", warning = FALSE,
               dependson = "setup")
@

<<sigdigits, cache = FALSE>>=
options(digits=3)
@

%%% CODE SETUP %%%

<<setup, dependson="">>=
library(factorial)

data(experimentDesign)
data(rnaseqMatrix)
data(microarrayMatrix)
data(microarrayRedMatrix)
data(microarrayGreenMatrix)
data(dubious)
@

<<process_matrices, dependson="setup">>=
# remove 0 count genes and genes classified as "dubious"
RS.filtered <- rnaseqMatrix[!(rowSums(rnaseqMatrix) == 0 |
                                            rownames(rnaseqMatrix) %in% dubious), ]

sharedGenes <- intersect(rownames(microarrayMatrix), rownames(RS.filtered))
RS.filtered <- RS.filtered[sharedGenes, ]
MA.filtered <- microarrayMatrix[sharedGenes, ]
MA.red.filtered <- microarrayRedMatrix[sharedGenes, ]
MA.green.filtered <- microarrayGreenMatrix[sharedGenes, ]

RG <- new("RGList", list(R=MA.red.filtered, G=MA.green.filtered))
@

<<normalize, dependson="process_matrices">>=
library(limma)
library(plyr)
library(dplyr)

matrices <- list(RS=RS.filtered, MA=RG)
raws = list(RS=RS.filtered, MA=MA.red.filtered)

subdesign.m = with(experimentDesign, cbind(Full=1,
                                           model.matrix(~ 0 + preparation),
                                           model.matrix(~ 0 + lane),
                                           model.matrix(~ 0 + preparation:lane))) == 1

# construct one row for each subset and each technology
# do_row and inflate are utility functions from factorial package
subsets <- data_frame(sub = colnames(subdesign.m)) %>%
    do_row(bool = subdesign.m[, sub]) %>%
    inflate(technology = c("MA", "RS")) %>%
    do_row(intensity = rowMeans(raws[[technology]][, bool])) %>%
    do_row(matrix = matrices[[technology]][, bool]) %>%
    do_row(pooled = pool_matrix(matrices[[technology]], experimentDesign, bool)) %>%
    inflate(normalize.method = c("none", "scale", "quantile", "cyclicloess")) %>%
    do_row(normalized = normalizeMatrix(pooled, normalize.method))
@

<<save_submatrices, dependson = "normalize">>=
submatrices <- subsets %>% dplyr::select(sub, technology, matrix, pooled) %>%
    filter(technology == "RS")

save(submatrices, file = "../data/submatrices.rda")
@

<<lmFit_ebayes, dependson="normalize">>=
pooled.design <- data.frame(condition = c("E", "E", "G", "G"))
mm <- model.matrix(~ condition, pooled.design)

subsets <- subsets %>%
    do_row(fit = lmFit(normalized, mm)) %>%
    do_row(eb = eBayes(fit))
@

<<tidied_DE, dependson="lmFit_ebayes">>=
# use biobroom package to turn into tidy data frame with p-values
library(biobroom)
tidied <- subsets %>%
    do_row(cbind(tidy(eb) %>% filter(term == "conditionG"), intensityraw = intensity)) %>%
    dplyr::select(-term)

# add intensity and significance rank
tidied <- tidied %>% dplyr::mutate(intensity = rank(intensityraw) / n(),
                                   significance = rank(p.value) / n())
@

<<DESeq2, dependson = "normalize">>=
# apply DESeq2 to RNA-Seq
library(DESeq2)

subsets_DESeq2 <- subsets %>% filter(normalize.method == "scale", technology == "RS")
perform_DESeq2 <- function(m) {
    dds <- DESeqDataSetFromMatrix(m, colData = pooled.design, design = ~ condition)
    suppressMessages(DESeq(dds))
}

subsets_DESeq2$dds <- lapply(subsets_DESeq2$pooled, perform_DESeq2)
@

<<DESeq2_tidied, dependson = "DESeq2">>=
library(biobroom)
DESeq2_tidied <- subsets_DESeq2 %>%
    group_by(normalize.method, technology, sub) %>%
    do_row(cbind(tidy(dds) %>% filter(term == "conditionG"),
             intensityraw = intensity))
@

<<tidy_significant, dependson="tidied_DE">>=
library(qvalue)
tidied <- tidied %>% mutate(q.value = qvalue(p.value)$qvalues)
summarized.sig <- tidied %>% summarize(significant = sum(q.value < .05),
                                       pi0 = pi0est(p.value)$pi0)
@

%% work with the full dataset, narrowing down to the subset we're doing

<<merged, dependson="tidy_significant">>=
# combine with pathway
data(pathway)
tidied <- merge(tidied, pathway, by.x="gene", by.y="ORF", all=TRUE) %>%
    mutate(Process = ifelse(is.na(Process), "All Other Genes",
                            as.character(Process))) %>%
    tbl_df()

# merging RS and MA the tidyr way: gather, unite, spread
# also add in column for intensity
library(tidyr)
merged <- tidied %>% gather(metric, value, estimate:q.value) %>%
    unite(techmet, technology, metric, sep=".") %>%
    spread(techmet, value) %>%
    mutate(intensity = pmin(MA.intensity, RS.intensity))
@

<<fulldata, dependson="merged">>=
# separate out the full (as opposed to subset) data
# also fix Process column to say "All Other Genes" and have counts
fulldata <- merged %>% filter(sub == "Full", normalize.method == "quantile") %>%
    dplyr::select(-sub) %>%
    arrange(Process) %>% tbl_df() %>%
    mutate(Process = add_counts(factor(Process)))
@

<<merged_pathway, dependson = "fulldata">>=
fulldata_pathway <- fulldata %>% filter(!is.na(Product))
@

%% rolling correlations
<<rolling_correlations, dependson = "merged">>=
library(TTR)

window.size <- 500

rollSpear <- function(x1, x2, n) {
    c(rep(NA, n - 1),
      rollapply(1:length(x1), n, function(i) {
          cor(x1[i], x2[i], method = "spearman")
          }))
}

merged_cors <- fulldata %>% arrange(normalize.method, intensity) %>%
    mutate(Pearson = runCor(MA.estimate, RS.estimate, n = window.size),
           Spearman = rollSpear(MA.estimate, RS.estimate, n = window.size))

cors_m <- merged_cors %>% dplyr::select(gene, intensity, Pearson, Spearman) %>%
    gather(Type, value, Pearson, Spearman)

overall_cors <- merged_cors %>% inflate(Type = c("Pearson", "Spearman")) %>%
    group_by(Type) %>%
    summarize(value = cor(MA.estimate, RS.estimate, method = tolower(Type[1])))
@

%%% GSEAMA %%%

<<GSEAMA>>=
library(GSEAMA)
library(org.Sc.sgd.db)

mm <- GOMembershipMatrix(org.Sc.sgdGO, min.size = 5, max.size = 250)
@

<<GSEAMA_run, dependson="GSEAMA">>=
merged_metrics <- fulldata %>%
    dplyr::select(gene, MA.p.value, RS.p.value, MA.estimate, RS.estimate) %>%
    gather(techmet, value, -gene) %>%
    separate(techmet, c("technology", "ignore", "metric"), c(2, 3)) %>%
    dplyr::select(-ignore)

test_association <- function(metric, genes) {
    # if it's p-values, use an alternative hypothesis of "less"
    alternative <- ifelse(all(metric >= 0 & metric <= 1), "less", "two.sided")
    # using "less" may have strange side effects- looking into it
	# alternative = "two.sided"
	TestAssociation(mm, genes, metric, method="wilcoxon", alternative=alternative)
}

GO_tab = toTable(org.Sc.sgdGO)

gseama_objs <- merged_metrics %>% group_by(technology, metric) %>%
    do(gseama = test_association(.$value, .$gene))

wilcox_sets <- gseama_objs %>%
    do_row(.$gseama[[1]]@colData) %>%
    mutate(Ontology = GO_tab$Ontology[match(ID, GO_tab$go_id)])

wilcox_genes <- gseama_objs %>% do_row(.$gseama[[1]]@geneData)

# get top sets using each method (is grouped by tech + metric)
top_sets <- wilcox_sets %>% filter(rank(pvalue) <= 15) %>%
    arrange(technology, metric, pvalue)

# arrange p-values for each column
set_pvalues <- wilcox_sets %>% unite(techmet, technology:metric, sep=".") %>%
    spread(techmet, pvalue)

top_sets_100 <- wilcox_sets %>% unite(techmet, technology:metric, sep=".") %>%
    filter(rank(pvalue) <= 100) %>%
    dplyr::select(techmet, pvalue, ID) %>%
    spread(techmet, pvalue)
@

<<wilcox_sets_depth, dependson = "GSEAMA_run">>=
median_per_set <- function(g, vec, dat) {
    vec <- vec[rownames(g@matrix)]
    m <- as.matrix(g@matrix) * vec
    m[m == 0] <- NA
    median_per_set <- apply(m, 2, function(x) median(x, na.rm = TRUE))

    unrowname(data.frame(ID = g@colData$ID, median = median_per_set))
}

g1 <- gseama_objs$gseama[[4]]
MA.dat <- cbind(median_per_set(g1, rowMeans(MA.red.filtered)),
                technology = "MA")
g2 <- gseama_objs$gseama[[4]]
RS.dat <- cbind(median_per_set(g2, rowSums(RS.filtered)),
                technology = "RS")
set_intensities <- rbind(MA.dat, RS.dat)

wilcox_sets <- wilcox_sets %>% inner_join(set_intensities)
#x <- reads_per_gene_per_set
wilcox_sets <- wilcox_sets %>% group_by(technology, metric) %>%
    mutate(intensity = rank(median) / n()) %>%
    mutate(bin = cut(intensity, seq(0, 1, 1 / 5),
                        include.lowest = TRUE))
@

% classify low intensity genes (compression vs exaggeration of each kind)

<<classify_low_intensity, dependson = "fulldata">>=
classify <- function(MA, RS, MAsig, RSsig) {
    ifelse(abs(MA - RS) < .5 | (!MAsig & !RSsig), "Similar",
           ifelse(sign(MA) != sign(RS) & (MAsig & RSsig), "Opposite Direction",
                  ifelse(MAsig & !RSsig, "MA only",
                         ifelse(!MAsig & RSsig, "RS only",
                                ifelse(abs(MA) > abs(RS), "MA Larger", "RS Larger")))))
}

cat_FDR <- .05

extremes <- fulldata %>%
    mutate(bin = ifelse(MA.intensity < .2 & RS.intensity < .2, "Low",
                        ifelse(MA.intensity > .9 & RS.intensity > .9,
                               "High", "Middle"))) %>%
    filter(bin != "Middle")

extremes <- extremes %>%
    mutate(MAsig = MA.q.value < cat_FDR, RSsig = RS.q.value < cat_FDR) %>%
    mutate(Greater = ifelse(abs(MA.estimate) > abs(RS.estimate), "MA", "RS")) %>%
    mutate(class = classify(MA.estimate, RS.estimate, MAsig, RSsig))

extremes %>% count(bin, class) %>% mutate(percent = n / sum(n)) %>%
    group_by() %>% dplyr::select(-n) %>% spread(bin, percent)

# extremes %>%

# ggplot(extremes, aes(MA.estimate, RS.estimate, color = class)) + geom_point() + geom_abline(col = "red") + facet_wrap(~ bin)
@

%% qPCR

<<qPCR, dependson = "setup">>=
library(plyr)
library(dplyr)

data(qPCR)

qPCR_DE <- qPCR %>% group_by(gene, symbol) %>%
    summarize(QP.estimate = mean(ct[condition == "e"]) -
                  mean(ct[condition == "g"]))

fulldata_qPCR <- fulldata %>% inner_join(qPCR_DE, by = "gene")

qPCR_comparisons <- fulldata_qPCR %>%
    dplyr::select(gene, symbol, MA.estimate, RS.estimate, QP.estimate) %>%
    tidyr::gather(technology, other, MA.estimate, RS.estimate) %>%
    mutate(technology = ifelse(technology == "MA.estimate", "Microarray", "RNA-Seq")) %>%
    mutate(residual = other - QP.estimate)
@


<<qPCR_summarized, dependson = "qPCR">>=
qPCR_summarized <- qPCR_comparisons %>% group_by(technology) %>%
    summarize(cor = cor(QP.estimate, other),
              p.value = cor.test(QP.estimate, other)$p.value,
              MSE = mean(residual^2))

qPCR_abs_residuals <- qPCR_comparisons %>%
    mutate(abs_residual = abs(residual)) %>%
    dplyr::select(-QP.estimate, -other, -residual) %>%
    mutate(technology = revalue(technology, c("RNA-Seq" = "RNASeq"))) %>%
    tidyr::spread(technology, abs_residual) %>%
    mutate(MA_RS_difference = Microarray - RNASeq)
#sum(qPCR_abs_residuals$MA_RS_difference > 0)
#sum(qPCR_abs_residuals$MA_RS_difference < 0)
@

<<utility>>=
# useful utility functions
millions <- function(x) round(x  / 1e6, 1)

corthreshold <- function(threshold) {
    m <- fulldata[fulldata$intensity > threshold, ]
    cor(m$MA.estimate, m$RS.estimate)
}
@

\begin{abstract}
Understanding the differences between microarray and RNA-Seq technologies for measuring gene expression is necessary for informed design of experiments and choice of data analysis methods. Previous comparisons have come to sometimes contradictory conclusions, which we suggest result from a lack of attention to the intensity-dependent nature of variation generated by the technologies. To examine this trend, we carried out a parallel nested experiment performed simultaneously on the two technologies that systematically split variation into four stages (treatment, biological variation, library preparation, and chip/lane noise), allowing a separation and comparison of the sources of variation in a well-controlled cellular system, \textit{Saccharomyces cerevisiae}. With this novel dataset, we demonstrate that power and accuracy are more dependent on per-gene read depth in RNA-Seq than they are on fluorescence intensity in microarrays. However, we carried out qPCR validations which indicate that microarrays may demonstrate greater systematic bias in low-intensity genes than in RNA-seq.
\end{abstract}

% \noindent {\bf A typical gene expression study consists of a pipeline involving several steps, including sample treatment, handling and library preparation, which may all contribute to variation observed in the data, in addition to the biological sources of variation that are of interest. These study-specific factors are all potential sources of variation or bias, which if ignored may severely mislead analysis and inference \cite{mecham2010supervised,Leek:2010jq}. We sought to examine these sources of variation and determine the fundamental quantitative differences between microarray and RNA-Seq, in order to improve methods of analysis and experiment design for each technology. We used an nested experiment performed simulataneously on RNA-Seq and microarrays, allowing unbiased separation and comparison of the sources of variation in a well-controlled and well-understood cellular system. With this novel dataset, we demonstrate that power and accuracy is more dependent on per-gene depth in RNA-Seq than on intensity in microarrays, but.}

\

\section{INTRODUCTION}

Since the introduction of RNA sequencing (RNA-Seq) for measuring mRNA expression, one important question has been how the technology compares to microarrays in power and accuracy. Experiments have been carried out to compare microarrays and RNA-Seq, with some concluding that RNA-Seq shows a greater power, accuracy and dynamic range \cite{tHoen:2008hn,Zhao:2014gb} and others challenging that conclusion \cite{Willenbrock:2009jv,McIntyre:2011fn}.

\begin{figure}[t]
\begin{center}
\includegraphics[width=\maxwidth]{images/figure_1.png}
\end{center}
\caption{Schematic of the experiment. The Condition and Biological Replicate steps were performed irrespective of technology and these materials were then utilized in a technology-specific manner (microarrays or RNA-seq) for the Preparation and Chip/Lane steps.\label{fig:schematic}}
\end{figure}

We carried out a genome-wide gene expression experiment in a controlled setting on the yeast \textit{Saccharomyces cerevisiae} in such a manner that the major sources of profiling variation can be unbiasedly partitioned and quantified (\textbf{Figure \ref{fig:schematic}}).  A single extraction of mRNA from each sample was quantified by both microarrays and RNA-seq in parallel.  We multiplexed each lane of RNA-seq profiling so that it exacty mirrored the eight chip per array design of the microarray platform that we utilized.  This experiment allowed a direct and completely parallel investigation into the quantitative operating characteristics of RNA-seq gene expression profiling versus microarrays.

We use this experiment to examine the variation added by the RNA-Seq and microarray technologies. Our analysis focuses on two novel problems: decomposing the variation contributed by each technology into multiple stages, which is made possible by the nested design, and analyzing the variation as a function of gene expression intensity, which is known to influence technology-specific variation in both microarrays \cite{Novak:2002dw,Tu:2002hr} and RNA-Seq \cite{Labaj:2011co,Tarazona:2011fi}. (Throughout this paper, we utilize ``intensity'' of a gene as a term for both fluorescence intensity in microarrays and per-gene read depth in RNA-seq, both of which are measurements of a gene's abundance subject to technology-specific biases and sources of variation.) We find that the variance contributed by RNA-Seq is more intensity-dependent than that from microarrays, a result that is statistically significant and is robust across multiple normalization methods and $R^2$ metrics. However, comparisons to qPCR validations show that microarray may show systematic biases in low intensities, possibly due to cross-hybridization. This has implications for the design of future experiments.

\ifNAR
  A novel characteristic of our experiment relative to previous microarray to RNA-Seq comparisons is that we utilized barcode multiplexing to combine RNA-Seq replicates on each
  % this manual page break is a huge hack, but it's necessary because otherwise it overflows into the bottom of the page
  \clearpage
  \noindent of two sequencing lanes, meaning that technical variation added by library preparation and handling is now distinguishable from ``sampling'' variation added by the lane. Also notable is that our analysis takes into account the effect that intensity has on variation, which has confounded previous comparisons \cite{Bullard:2010p2947}. We thus examine which technology adds more variation as a function of per-gene depth or intensity.
\else
  A novel characteristic of our experiment relative to previous microarray to RNA-Seq comparisons is that we utilized barcode multiplexing to combine RNA-Seq replicates on each of two sequencing lanes, meaning that technical variation added by library preparation and handling is now distinguishable from ``sampling'' variation added by the lane. Also notable is that our analysis takes into account the effect that intensity has on variation, which has confounded previous comparisons \cite{Bullard:2010p2947}. We thus examine which technology adds more variation as a function of per-gene depth or intensity.
\fi

\section{MATERIALS AND METHODS}

\subsection{Yeast growth and gene expression profiling}

\subsubsection{Growth of yeast in chemostats.}

The experiment was performed using yeast haploid strain DBY 12000 FY, a S288C derivative containing the wild-type HAP1 allele.  A single colony was split into 2 overnight cultures, one containing ethanol limitation medium with 60 C-mM and the other containing glucose limitation medium with 27 C-mM Carbon as the carbon source. 1 mL of each culture was used to inoculate a chemostat. After the chemostats reached stasis, 10 ml culture was collected and frozen at -80C (see \textbf{Supplementary Methods}).  A second set of biological samples were prepared the same way on a different day.

\subsubsection{Microarray and RNA-Seq.}

RNA was harvested from each of the 4 samples. The 4 RNA samples were then processed twice on different days using the Agilent Quick Amp Labeling kit (Part no. 5190-0424) to produce 8 cRNA libraries, each of which was then hybridized on two separate chips (Yeast Expression 8x15K arrays) on different days according to the factorial design.  After the arrays are washed and scanned, features are extracted using the Agilent Feature Extraction software to determine red and green intensities.

The same 4 RNA samples were also processed using the Illumina TruSeq RNA Sample Prep v2 LS protocol.  Each sample was prepared twice up to the 3' end adenylation, then each of the 8 preparations was split into two aliquots, after which each was indexed and amplified to complete the RNA-Seq library preparation. The two groups of 8 samples that were indexed together were each mixed at equimolar concentrations and sequenced on separate lanes on the same Illumina HiSeq 2500 flowcell, to produce 141bp reads. In total, we produced profiles from 16 RNA-Seq samples and 16 microarray samples with identically nested experimental designs.

\subsubsection{Quantitative PCR.}

To perform quantitative PCR, the four RNA samples were treated with RNase-Free DNase Set  (Cat \# 79254, Qiagen) column digestion to remove DNA in the samples. Total RNA was quantified using Quant-iT RNA assay Kit (Q33140, Invirogen) and Biotek Synergy Mx plate reader. 750 ng total RNA was used in the first strand cDNA synthesis (SuperScript III Reverse Transcriptase, Cat \# 18080, Invitrogen). Prevalidated FAM-MGB Taqman probes and primers mix for all candidate genes were ordered from Life Technologies. 96 well plates (Cat\# N8010560, ABI) and optical adhesive cover starter kit (Cat \# 4313663, ABI) were used. qPCR reactions were set up by combining Taqman Gene Expression Master Mix (Cat \# 4369016, Life Technologies) and individual probe plus primer mix. 20 µl qPCR reaction was run on ABI 7900 HT Sequence Detection System using the following thermal protocol: 50°C, 2 min; 95°C, 10 min; 40 cycles of 95°C, 15 sec and 60°C, 1 min; 95°C, 15 sec; 60°C, 15 sec; 95°C, 15 sec.

Any candidate genes with more than one band on agarose gel after qPCR were excluded from further analysis. Three replicates of RT-qPCR for each RNA sample starting from cDNA synthesis to qPCR reaction were performed. Each measurement was normalized based on the average across all genes in a biological replicate, and the log-fold change was estimated based on the difference in average number of cycles between ethanol and glucose samples.

<<limma_version, dependson = "lmFit_ebayes">>=
limma.version = sessionInfo()[["otherPkgs"]][["limma"]]$Version
@

\

\subsection{Preprocessing and statistical analysis}  We used \texttt{bowtie2} with the default set of parameters to map the RNA-Seq reads to the yeast genome, and used \texttt{htseq-count} to match reads against the \emph{S. cerevisiae} R64 release of the reference genome from the Sacchromyces Genome Database. Microarrays were normalized after averaging all preparation and chip replicates within each biological replicate to create two E replicates and two G replicates. The matrix of RNA-Seq counts was first pooled within preparation and chip replicates, then was transformed using \texttt{voom} from \texttt{limma}, which also computed precision weights for each observation \cite{Law:2014tta}. This pooling was necessary for differential expression because while the replicates at later nested stages introduced variation, they were not full biological replicates, and treating them as replicates for limma would be pseudoreplication that underestimates the within-group variation \cite{Hurlbert:1984cy}. We tested for differential expression in each technology using a linear model with empirical Bayes shrinkage of T-statistics, implemented by \texttt{limma} version \Sexpr{limma.version} \cite{Smyth:2005wm}.

\

\subsubsection{Estimating contributions to variation.}

For calculations of the percent of variation explained, the microarray red/green log fold changes and the RNA-Seq log-transformed counts were compared, with the counts first transformed using \texttt{voom} \cite{Law:2014tta}. We compared multiple methods of normalizing both the microarray and RNA-Seq data between samples, using implementations in the \texttt{normalizeBetweenArrays} function in \texttt{limma} as well as the TMM \cite{Robinson:2010p9965} and RLE \cite{Anders:2010fua} methods for RNA-Seq, but none made a qualitative difference in the resulting conclusions (\textbf{Figure S3}). % \ref{fig:compare_normalization}}).

For each gene, in both the normalized RNA-Seq data or the microarray fold-change values, we calculated the adjusted-$R^2$ using a nested ANOVA on three linear models, which differed in which levels of the experiment were parameterized and which were left as residual variation (\textbf{Supplementary Methods}). The $R^2$ estimates were then smoothed using LOESS across all genes based on their microarray intensity or RNA-Seq read depth.

\section{RESULTS}

We carried out an experiment on both Agilent microarrays and Illumina RNA sequencing to investigate the effects of each source of variation on the inference of differential expression. We used the widely studied model organism \textit{Saccharomyces cerevisiae} to investigate differential expression associated with growth in different carbon sources, glucose (G) and ethanol (E), and we introduced steps to capture three additional factors or sources of variation. The sources of variation are: (1) {\bf Condition}: biological condition of interest (G vs. E); (2) {\bf Biological Variation}: natural biological variation between clonal populations; (3) {\bf Preparation}: sample handling and preparation; (4) {\bf Chip/Lane}: technical variation associated with either technology, such as array effects or lane effects.

Factors (1)--(4) were sequentially nested, and at the stage of each nested factor, the sample was split such that each sample from the previous factor is balanced across both levels of the factor (\textbf{Figure \ref{fig:schematic}}). Factors (1) and (2) were performed only once to produce four samples of isolated RNA, while factors (3) and (4) were technology-dependent and therefore performed in parallel with microarrays and with RNA-Seq. We took advantage of a similar design on Agilent yeast microarrays (8 hybridizations per chip) and Illumina RNA-Seq (8 indexed samples per lane) to mimic the same approach across the two technologies, resulting in 16 microarray and 16 RNA-Seq profiles that show the amount of variation added at each stage. The RNA-Seq experiment achieved a depth of \Sexpr{millions(sum(RS.filtered))} million reads, with depths of \Sexpr{millions(sum(RS.filtered[, experimentDesign$lane == 1]))} million and \Sexpr{millions(sum(RS.filtered[, experimentDesign$lane == 2]))} million on each of the two lanes.

\

\subsection{Differential expression} The biological goal of this experiment is to infer differential gene expression in \textit{Saccharomyces cerevisiae} strain DBY12000 (S288c Hap1+ Mat \textit{a}) cultivated in balanced growth conditions in chemostats using either glucose or ethanol as the sole carbon source (condition of interest).  The chemostat device helped minimize any variations in growth conditions (such as physiological state, temperature, nutrient composition, etc.) so the study could directly interrogate the factor of interest: transcriptional responses to different carbon sources \cite{Novick:1950va}. We used a linear model with t-statistic shrinkage to detect differential expression in each case, after log-transforming the RNA-Seq counts and fitting precision weights to each observation \cite{Smyth:2005wm,Law:2014tta}. We specifically tracked a biologically relevant set of \Sexpr{nrow(pathway)} genes known to be involved in processes relevant to glucose or ethanol metabolism, namely gluconeogenesis, glycolysis, the TCA cycle and the pyruvate branchpoint \cite{Kolkman:2005tz}.

<<cortypes, dependson = "calculate_rolling_correlations">>=
pearson_low <- min(filter(cors_m, Type == "Pearson", !is.na(value))$value)
pearson_high <- max(filter(cors_m, Type == "Pearson", !is.na(value))$value)
spearman_low <- min(filter(cors_m, Type == "Spearman", !is.na(value))$value)
spearman_high <- max(filter(cors_m, Type == "Spearman", !is.na(value))$value)
@


<<color_scale>>=
# hardcode scale: items are alphabetical
library(ggplot2)
color_order <- c("black", "blue", "lightblue", "yellow", "green")
color_scale <- scale_color_manual(values=color_order)
@

\begin{figure}
<<comparefig, dependson = c("fulldata", "color_scale")>>=
library(ggplot2)

comparefig <- ggplot(fulldata, aes(RS.estimate, MA.estimate, color = Process, alpha = intensity)) +
    geom_point() + geom_abline(col="red") +
    xlab(expression(atop(paste("RNA-Seq ", log[2], "(G/E)")))) + ylab(expression(atop(paste("Microarray ", log[2], "(G/E)")))) + color_scale
@

<<figure_2, dependson="calculate_rolling_correlations">>=
library(ggplot2)

rollcor <- ggplot(cors_m) +
    geom_line(aes(intensity, value, color = Type)) +
    geom_hline(aes(yintercept = value, color = Type), data = overall_cors, lty = 2) +
    xlab("Quantile of MA intensity or RS depth") +
    ylab(expression(atop(paste("Correlation of ", log[2], "(G/E) estimates"), "between MA and RS")))

library(gridExtra)
g.a = textGrob(.5, unit(1,"npc") - unit(1,"line"), label="a)")
g.b = textGrob(.5, unit(1,"npc") - unit(1,"line"), label="b)")
print(arrangeGrob(g.a, comparefig + coord_fixed(), g.b, rollcor, nrow=2, widths=c(.03, .97)))
@
\caption{a) Comparison between the log$_2$(G/E) (log fold-change) estimates measured with RNA-Seq or with the microarray. The transparency of the points corresponds to the quantile of the intensity in microarray or RNA-Seq, whichever is lower. \Sexpr{NROW(pathway)} genes from biologically relevant pathways are highlighted in color. b) Pearson or Spearman correlation of log$_2$(G/E) estimate between microarray and RNA-Seq, within a \Sexpr{window.size} gene window rolling over microarray/RNA-Seq intensity ranking. The correlation of all genes (\Sexpr{overall_cors$value[1]} Pearson, \Sexpr{overall_cors$value[2]} Spearman) is shown as a horizontal dashed line. \label{fig:logFC_comparison}}
\end{figure}

One goal is to assess to what extent RNA-Seq and microarray experiments agree in their assessment of differential expression. \textbf{Figure \ref{fig:logFC_comparison}a} compares the estimated log$_2$(G/E) fold change ratios between the two technologies, with the opacity of each point determined by the quantile of the microarray intensity or RNA-Seq read depth, whichever is lower. The microarray intensity of each gene was calculated as the average cy5 flourescence intensity across all samples (where cy5 is the channel corresponding to the samples of interest), while RNA-Seq depth was calculated as the total number of reads mapping to the gene across all samples. We also show \Sexpr{nrow(pathway)} biologically relevant genes highlighted in color. Overall the two technologies show a correlation of \Sexpr{corthreshold(0)}, but this correlation is highly dependent on read depth and microarray intensity, with the lowest-intensity genes exhibiting the greatest noise and therefore the lowest correlation.

\textbf{Figure \ref{fig:logFC_comparison}b} shows how the Pearson and Spearman correlations between microarray and RNA-Seq effect size estimates depend on per-gene intensity, using a rolling window of \Sexpr{window.size} genes, ordered by intensity (again determined by the quantile of the microarray intensity or RNA-Seq read depth, whichever is lower). The Pearson correlation varies from \Sexpr{pearson_low} to \Sexpr{pearson_high}, while the Spearman ranges from \Sexpr{spearman_low} to \Sexpr{spearman_high}. The \Sexpr{nrow(pathway)} genes in our biologically relevant set showed a \Sexpr{cor(fulldata$MA.estimate[!is.na(fulldata$Product)], fulldata$RS.estimate[!is.na(fulldata$Product)])} correlation of effect size estimates, which is understandable since almost all lie in the top 10\% of read depth and microarray intensity. The microarray and RNA-Seq assays identified as significantly differentially expressed \Sexpr{sum(fulldata_pathway$MA.q.value <= .05)} of the \Sexpr{nrow(pathway)} biologically relevant genes at estimated FDR $\leq 5\%$ \cite{ST03}, and the two technologies agreed on the direction of the change for all of these genes. This suggests that there is little difference between the two technologies in terms of estimating differential expression of high-intensity genes.

\

\subsection{Percentage of variation explained}

The primary goal of the factorial experiment is to determine the relative amount of variation added at each stage of the experiment for each of the two technologies. Based on the correlation matrix (\textbf{Figure S1}), the RNA-Seq and microarray assays easily distinguished between the ethanol and glucose samples, and showed clustering within the chip/lane replicates, as expected from the experimental design. Any analysis should, however, consider the intensity-dependence of the variation that each technology contributes. We calculated the proportion of variation explained by the Condition, Biological, and Preparation factors, as well as the residual variation due to the chip or lane, using a nested ANOVA analysis (\textbf{Materials and Methods}). We computed this breakdown separately for each gene and smoothed the result across microarray intensity RNA-Seq depth using the local regression method, LOESS (\textbf{Figure \ref{fig:percentvar}}).

%% R^2 work

<<R2s, dependson="normalize">>=
# calculating R2s also happens in a tidy framework, but only for normalization
# methods and technologies (not subsets as happen for DE)
library(methods)

normalize.methods <- c("none", "scale", "quantile", "cyclicloess", "TMM", "RLE")

geneR2s <- expand.grid(technology = c("RS", "MA"),
                       normalize.method = normalize.methods,
                       stringsAsFactors = FALSE)

geneR2s <- geneR2s %>% group_by(technology, normalize.method) %>%
    do(matrix = normalizeMatrix(matrices[[.$technology[[1]]]],
                          normalize.method=.$normalize.method[[1]]))

geneR2s <- geneR2s[!sapply(geneR2s$matrix, is.null), ]

geneR2s <- geneR2s %>% group_by(technology, normalize.method) %>%
    do(construct_R2s(as.matrix(.$matrix[[1]]), rowMeans(raws[[.$technology[1]]])))
@

<<process_R2s, dependson="R2s">>=
geneR2s <- geneR2s %>% group_by(technology, normalize.method, Level) %>%
    dplyr::mutate(Quantile = rank(Intensity) / n()) %>%
    filter(!is.na(value), !is.infinite(value)) %>%
    group_by() %>%
    mutate(normalize.method = factor(normalize.method, levels = normalize.methods))
geneR2s_quantile <- geneR2s %>% group_by() %>% filter(normalize.method == "quantile")
@

\begin{figure}
<<figure_3, dependson="process_R2s">>=
g <- ggplot(geneR2s_quantile, aes(Quantile, value, color=Level, lty=technology)) +
    geom_smooth(method="loess", se=FALSE) +
    labs(lty = "Technology") +
    xlab("Quantile of microarray intensity or RNA-Seq read depth") +
    ylab("Smoothed % of variation explained")
g
@
\caption{Percent of variance explained by each nested level of the experiment as computed by an ANOVA adjusted $R^2$, smoothed using LOESS across the intensity quantiles.\label{fig:percentvar}}
\end{figure}

In both technologies at almost all intensities, the largest sources of variation were the treatment (ethanol versus glucose) and the chip/lane, in a tradeoff that depended strongly on intensity or read depth. The analysis indicated that the variance due to RNA-Seq lane at low-intensity genes was greater than that due to microarray chip. This difference is statistically significant: when the genes are divided into 10 bins based on depth or intensity, all bins in the 80\% lowest intensity show a highly significant difference between the amount of variation added by the microarray chip versus the sequencing lane (\textbf{Figure S2}). These conclusions were robust across multiple methods of microarray and RNA-Seq normalization (\textbf{Figure S3}). To ensure that this discrepancy did not arise from the difference between the discrete count data from RNA-Seq and the quasi-continuous fluroescence intensity data from the microarray, we also calculated an alternative $R^2$ designed for count data to determine the proportions of variance explained (\textbf{Supplementary Methods}) \cite{ColinCameron:1997iv}, and observed almost no difference (\textbf{Figure S4}).


%% compare within technology

<<tidied_1lane_1prep, dependson="tidied_DE">>=
td_combined <- DESeq2_tidied %>% group_by() %>%
    mutate(normalize.method = "DESeq2") %>% filter(!is.na(estimate)) %>%
    rbind_list(tidied)

tidied.1lane.1prep <- td_combined %>% filter(!grepl(":", sub) & sub != "Full") %>%
    separate(sub, c("level", "replicate"), -2) %>%
    mutate(replicate = ifelse(replicate %in% c("1", "A"), "rep1", "rep2"))

# lane to lane and prep to prep comparisons
tidied.coefs <- tidied.1lane.1prep %>%
    dplyr::select(technology, normalize.method, level, gene, replicate, estimate) %>%
    spread(replicate, estimate) %>%
    mutate(absdiff = abs(rep2 - rep1))

# use the intensity from the full data
fullintensity <- tidied %>%
    filter(sub == "Full", normalize.method == "quantile") %>% group_by() %>%
    dplyr::select(gene, technology, intensity)

tidied.coefs <- tidied.coefs %>% inner_join(fullintensity) %>%
    filter(normalize.method %in% c("DESeq2", "quantile")) %>%
    group_by() %>%
    mutate(normalize.method = plyr::revalue(normalize.method, c("quantile" = "limma")))
@

\begin{figure}
<<calculate_rolling_correlations, dependson="tidied_1lane_1prep">>=
library(TTR)

window.size <- 500

tidied.coefs <- tidied.coefs %>% arrange(technology, normalize.method, level, intensity) %>%
    group_by(technology, level, normalize.method) %>%
    filter(!is.na(rep1), !is.na(rep2), level == "lane") %>%
    mutate(correlation=runCor(rep1, rep2, n = window.size))

# draw only every nth point to keep it smooth
correlation_fig <- tidied.coefs %>% filter(level == "lane") %>%
    dplyr::slice(seq(1, n(), 25)) %>%
    ggplot(aes(intensity, correlation, lty = technology, color = normalize.method)) +
    geom_line() + scale_linetype_manual(values = c(1, 2))
@

<<absdiff_fig, dependson = "rolling_correlations">>=
absdiff_fig <- ggplot(tidied.coefs, aes(intensity, absdiff, lty = technology, color = normalize.method)) +
    geom_smooth(method = "loess", se = FALSE) +
    scale_linetype_manual(values = c(1, 3))
@

<<figure_4, dependson = c("rolling_correlations", "absdiff_fig")>>=
library(gridExtra)

#https://github.com/hadley/ggplot2/wiki/Share-a-legend-between-two-ggplot2-graphs
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}

g.a = textGrob(.5, unit(1,"npc") - unit(1,"line"), label="a)")
g.b = textGrob(.5, unit(1,"npc") - unit(1,"line"), label="b)")
leg <- g_legend(correlation_fig + labs(lty = "Technology", color = "Method"))

f1 <- correlation_fig + theme(legend.position = "none") +
    xlab("Quantile of microarray intensity or RNA-Seq depth") +
    ylab(expression(paste("Correlation of ", log[2], "(G/E) btwn lanes/chips")))

f2 <- absdiff_fig + theme(legend.position = "none") +
    xlab("Quantile of microarray intensity or RNA-Seq depth") +
    ylab(expression(paste("Abs. difference of ", log[2], "(G/E) btwn lanes/chips")))

leftside <- arrangeGrob(g.a, f1, g.b, f2,
                        nrow = 2, widths = c(.03, .97))
print(arrangeGrob(leftside, leg, nrow = 1, widths = c(.8, .2)))
@
\caption{Comparisons of the $\log_2$(G/E) estimates within each technology, comparing the estimates computed for each of the two chips in a microarray or two lanes in RNA-Seq, using the same library preparations. a) Pearson correlation between the two lanes of RNA-Seq or two chips of microarrays, within a \Sexpr{window.size} gene window rolling over intensity. b) Absolute value of the $\log_2$(G/E) estimate difference between chips in microarray or lanes in RNA-Seq, smoothed using LOESS. \label{fig:rollcor}}
\end{figure}

To examine the variation added by the chip/lane level more directly, we also measured the difference in log fold change estimate when the same sample was run on two chips or two lanes. We computed the Pearson and Spearman correlation of log fold changes between chips or between lanes in overlapping windows of \Sexpr{window.size} genes ordered by intensity as above (\textbf{Figure \ref{fig:rollcor}a}), as well as a LOESS-smoothed curve of the absolute value of the difference (\textbf{Figure \ref{fig:rollcor}b}). Both analyses confirm that the difference between RNA-Seq lanes is more intensity-dependent than the difference between microarray chips, with a particularly great disagreement in low-intensity genes. One notable question is whether this effect can be mitigated by effect size shrinkage, such as the DESeq2 software, which is designed to improve the stability of estimates for low-depth genes \cite{Love:2014vr}. \textbf{Figure \ref{fig:rollcor}} shows that DESeq2 causes the absolute difference in log fold changes to decrease, but does not substantially improve the correlation in any but the lowest-intensity windows. This suggests that effect size shrinkage can make estimates less variable between lanes, but does not remove the intensity dependence.

<<qPCR_values, dependson = "qPCR">>=
nqPCR <- length(unique(qPCR_comparisons$gene))

qPCR_cors <- qPCR_comparisons %>% group_by(technology) %>%
    do(tidy(cor.test(.$QP.estimate, .$other)))

MA_QP <- qPCR_cors %>% filter(technology == "Microarray")
RS_QP <- qPCR_cors %>% filter(technology == "RNA-Seq")
@

\

\subsection{Validation of low-intensity genes}

These results do not necessarily indicate that microarrays are more accurate at low intensities than RNA-Seq, only that they show greater consistency between replicates. Each technology may still possess biases that cause their measurements not to reflect the underlying mRNA abundance levels. To examine the accuracy of each technology more directly, we chose \Sexpr{nqPCR} genes in the bottom 20\% of intensity for both technologies, for which the estimates of the log$_2$(G/E) fold change disagreed by at least 1.0 between the technologies. On these low-intensity contested genes, we performed quantitative PCR (qPCR) on the same mRNA as the experiment to provide an independent validation (\textbf{Materials and Methods}). As shown in \textbf{Figure \ref{fig:qPCR_scatter}}, the qPCR results agreed much more closely with the RNA-Seq estimates than with the microarray: the correlation of estimated RNA-Seq and qPCR log fold changes is \Sexpr{RS_QP$estimate} (p-value $\approx\Sexpr{format.pval(RS_QP$p.value)}$), while the microarray to qPCR correlation is \Sexpr{MA_QP$estimate} (p-value $\approx\Sexpr{format.pval(MA_QP$p.value)}$).% In only \Sexpr{sum(qPCR_abs_residuals$MA_RS_difference < 0)} out of \Sexpr{nqPCR} cases was the microarray estimate closer to the qPCR estimate than RNA-Seq was, and in no case was the improvement greater than $\Sexpr{-min(qPCR_abs_residuals$MA_RS_difference)}$.


\begin{figure}
<<figure_5, dependson = "qPCR", fig.height = 6, fig.width = 8>>=
library(ggplot2)
library(tidyr)

ggplot(qPCR_comparisons, aes(QP.estimate, other)) +
    geom_text(aes(label = symbol), hjust = 1, vjust = 1, size = 4) +
    geom_point() + facet_wrap(~ technology) +
    geom_abline() +
    ylab(expression(atop(paste("Microarray or RNA-Seq ", log[2], "(G/E) estimate")))) +
    xlab(expression(atop(paste("qPCR ", log[2], "(G/E) estimate")))) +
    xlim(-2, 2.5)
@
\caption{Comparison of the log fold change estimates measured with qPCR compared to estimates from the microarray or RNA-Seq, for \Sexpr{nqPCR} selected low-intensity genes that disagreed between RNA-Seq and microarray. \label{fig:qPCR_scatter}}
\end{figure}

This leads to the question of whether the disagreement between microarrays and qPCR results from greater variation present at low-intensity, or rather systematic bias introduced by the technology, such as cross-hybridization or fold-change compression. Towards that end, we examined the distribution of normalized measurements across all biological, technical and chip/lane replicates from the 6 genes for which microarrays most strongly disagreed with qPCR (\textbf{Figure \ref{fig:qPCR_boxplots}}). Even though these genes are low-intensity, in most cases the variation within microarray and RNA-Seq measurements was very small compared to the disagreement between the technologies. In a dramatic example, GAS2 shows a reversed fold change relative to qPCR and RNA-Seq, but shows very little within-condition variation in any technology. Of these 6 genes, only OSW1 could plausibly be caused by fold change compression, since in other cases the microarray effect was greater than or a reversal from the RNA-Seq and qPCR observations \cite{Ambroise:2011hb}. This suggests that the issue within these genes was not random variation present in low-intensity genes or fold change compression, but rather a bias that led to a spurious but statistically significant effect size estimate.

\begin{figure}
<<figure_6, dependson = c("lmFit_ebayes", "qPCR")>>=
# tidy normalized expression data (not just data)
library(biobroom)

levels <- c("condition", "biological", "technical", "lane")
normalized_tidy <- subsets %>%
    filter(sub == "Full") %>%
    group_by(normalize.method, technology, sub) %>%
    do(tidy(normalizeMatrix(.$matrix[[1]]))) %>%
    group_by() %>%
    separate(sample, levels, sep = 1:3)

qPCR_mod <- qPCR %>% mutate(condition = toupper(condition)) %>%
    mutate(value = -ct, technology = "QP") %>%
    dplyr::select(technology, condition, biological, technical, gene, value,
                  symbol)

# normalize
qPCR_mod <- qPCR_mod %>% group_by(technical) %>%
    mutate(value = value - mean(value))

norm_filt <- normalized_tidy %>%
    filter(sub == "Full", gene %in% qPCR_mod$gene) %>%
    dplyr::select(-sub, -normalize.method)

renamer <- c(MA = "Microarray", RS = "RNA-Seq", QP = "qPCR")
norm_combined <- rbind_all(list(norm_filt, qPCR_mod)) %>%
    mutate(technology = revalue(technology, renamer)) %>%
    group_by(technology, gene) %>%
    mutate(centered = value - mean(value),
           symbol = as.character(qPCR$symbol)[match(gene, qPCR$gene)])

# top differences
topdiffs <- (qPCR_abs_residuals %>% arrange(desc(MA_RS_difference)))$gene

num_qPCR_compare <- 6
ggplot(norm_combined %>% filter(gene %in% head(topdiffs, num_qPCR_compare)),
       aes(condition, centered)) + geom_boxplot() +
    facet_grid(symbol ~ technology, scales = "free_y") +
    ylab("Treatment") + ylab("Normalized and centered expression estimate")
@
\caption{Boxplots comparing the normalized and centered expression values of microarrays, RNA-Seq, and qPCR of the \Sexpr{num_qPCR_compare} genes for which microarray and qPCR most disagreed. This shows that in many cases, microarray measurements were very consistent between biological, technical and chip replicates. This suggests that the problem is not variation at low-intensity microarray measurements, but rather bias.\label{fig:qPCR_boxplots}}
\end{figure}


\

\subsection {Gene set enrichment}

Many gene expression analyses seek to determine sets of genes whose expression changes within a particular condition in order to draw biologically relevant conclusions \cite{Subramanian:2005jt,Ackermann:2009bw}. We thus looked for enriched sets of genes within each technology using a Wilcoxon rank-sum test comparing genes within a gene set to those outside it \cite{Gresham:2011iq,Hung:2012fp}.

Many enrichment tests look for genes that are unusually high or low on a ranked list, often ranked by statistical significance. However, ranking by statistical significance has been noted to lead to a confounding effect in RNA-Seq, where highly-expressed or high-depth gene sets are spuriously marked as significant \cite{Alicia:2009p2913,Gao:2011p9925}. An alternative is to perform enrichment analysis on log fold change estimates, which would be expected to be less confounded with depth \cite{Robinson:2014hz}. To demonstrate this effect in our data, we performed gene set enrichment using either p-values or fold change estimates from each technology, then assigned the gene sets into 5 bins based on each gene set's median intensity (\textbf{Figure S5}). We see that the enrichment p-value histogram is highly conservative for low-intensity gene sets when differential expression p-values are used, and is less influenced by intensity when the log fold change estimate is used instead. Notably, even though we found the per-gene variation to be more intensity dependent in RNA-Seq than in microarray, the intensity dependence of gene sets is similar between the two technologies. Another reason to use the fold-change estimate is that the significance metric is more technology dependent: the p-values for differential expression show only a moderate Spearman correlation of \Sexpr{cor(fulldata$RS.p.value, fulldata$MA.p.value, method="spearman")} between microarrays and RNA-Seq, while the fold change estimate shows a higher correlation of \Sexpr{cor(fulldata$RS.estimate, fulldata$MA.estimate, method="spearman")}. We thus chose to use the estimate of effect size rather than statistical significance to evaluate gene set enrichment.

<<eval = FALSE>>=
# \Sexpr{cor(wilcox.results[, 1], wilcox.results[, 2], method="spearman")}
#
@

The results of our gene set analysis of both the RNA-Seq and the microarray data are shown in Table S2, and the distributions of the log fold change of some of the most significantly enriched gene sets are shown in \textbf{Figures S6-S8}. The enrichment of gene sets for carbohydrate catabolic process and glycolysis in glucose and of mitochondrial respiratory chain and ATP synthesis coupled proton transport serve as confirmation that the experiment captures the difference between the two metabolic states. Two of the three most significantly enriched sets are cytoplasmic translation and mitochondrial translation, for which expression is higher in glucose and in ethanol, respectively. As the rate of respiration in the mitochondria is higher in ethanol than in glucose, this suggests that increase in mitochondrial activity is reflected in a tradeoff of translation from the cytoplasmic ribosomes to the mitochondria.  Another notable result is that genes in iron ion homeostasis and ferric-chelate reductase activity are higher expressed in ethanol than in glucose. This is likely due to the important role of iron transport and reduction in heme biosynthesis, which in turn is necessary for the electron transport chain and other respiratory activity \cite{Outten:2013dl,Morano:2012jy,Lange:1999ja}.

We identified higher expression of cytokinesis and cellular budding genes in ethanol, even though growth rate was kept equal between the two samples. Mitochondrial inheritance and distribution is known to be actively regulated by the budding tip and to be necessary for equal and efficient distribution of the organelles \cite{Simon:1997ep,Boldogh:2005gq}, and indeed some differentially expressed genes in our experiment, such as UTH1, have been identified as being related to both cell wall biogenesis and mitochondrial division \cite{Camougrand:2000jz,Velours:2002jb,Camougrand:2004bx}. Our results suggests that this process of mitochondrial inheritance may be transcriptionally regulated in response to the metabolic state or level of respiratory activity.

\

\section{DISCUSSION}

We demonstrate an experimental and statistical approach for determining the variation added at each stage of a microarray or RNA-Seq experiment. We determined that RNA-Seq shows a greater degree of intensity-dependent variation than do microarrays, with particularly high variance for low-intensity genes, and that the intensity-dependent component was contributed mostly by the chip or lane level. With qPCR validation, however, we discovered that microarrays appear to possess some systematic biases in their estimation of differential expression for low-intensity genes. Since this bias appears to be consistent across biological, technical and chip replicates, it likely cannot be solved or even detected by performing additional replicates on the microarray platform.

Our results have implications for the design of microarray and RNA-Seq experiments meant to identify differential expression. While other experiments will vary in the amount of variation added at the biological stages, that variation is likely to be intensity-independent as it was in our study, meaning our qualitative conclusions are likely to hold. For high-intensity genes there is little difference either in the genes called significant or the estimate of effect size between RNA-Seq and microarray, and therefore the decision of which technology can be made on other criteria, such as cost. However, in low-intensity genes, the RNA-Seq technology tends to add greater variation, leading to lower statistical power and greater uncertainty in expression estimates. Microarrays, while more consistent in their estimates across technical replicates, show systematic biases that confound differential expression detection. This suggests that studies for which low-expressed genes are of special interest should be performed cross-platform.

More importantly, our study has demonstrated that the intensity-dependent nature of variation must be taken into account in future technology comparisons and quality control experiments. Our approaches of dividing genes into intensity bins, observing correlations within overlapping windows, and smoothing per-gene values using LOESS showed trends and differences in the technologies that would have been missed using aggregate statistics. Finally, we have demonstrated how our experimental data is an appropriate benchmark for comparing statistical analysis methods and for developing experimental recommendations, as it analyzes a well-studied system, includes variation at each stage of an experiment, and compares RNA-Seq and microarrays directly on the same biological samples. We expect future research by ourselves and others will extend our conclusions and develop them further.

\section{Funding}

This work was supported by the National Institutes of Health [grant numbers R01 HG002913, R21 HG006769].

\section{Computer Code and Data}

The microarray data used in this experiment is available from the Gene Expression Omnibus (accession GSE65365), and the RNA-Seq reads are available from the NCBI Short Read Archive (BioProject accession PRJNA271248). Reproducible R code and the processed data can be obtained from \url{https://github.com/StoreyLab/factorial-MA-RS}.

\section{Acknowledgments}

We are grateful for the experimental guidance from Dr. Dipen Sangurdekar, Dr. Sandy Silverman and Dr. Nicolai Slavov.

\clearpage

%% Pseudo-R2

<<pseudo_R2, dependson=c("RNA_Seq_pooled", "analyze_microarray")>>=
library(ggplot2)
library(edgeR)
library(reshape2)

cond = experimentDesign$condition

d = DGEList(counts=RS.filtered, group=cond)
d <- calcNormFactors(d)

mm = model.matrix(~ condition + condition:biological + condition:biological:preparation, experimentDesign)

d = estimateGLMTrendedDisp(d, mm)
d = estimateGLMTagwiseDisp(d, mm)

logliks = function(m, pred, disp=FALSE) {
  # if dispersion is 0, use Poisson
  sapply(1:NROW(m), function(i) {
    if (length(disp) == 1 & disp[1] == 0) {
      return(sum(log(dpois(m[i, ], pred[i, ]))))
    }
    else {
      return(sum(log(dnbinom(m[i, ], mu=pred[i, ], size=1 / disp[i]))))
    }
  })
}

dfs = lapply(c(1, 2, 4, 8), function(k) 1:k)
design.matrices = c(lapply(dfs, function(k) mm[, k]))
design.matrices[[1]] = matrix(design.matrices[[1]])

n = NCOL(RS.filtered)
p = c(1, 2, 4, 8)

RNA.Seq.pseudoR2s = function(disp, method) {
    #fit = glmFit(d, mm, dispersion=disp)
    fits = sapply(design.matrices, function(m) glmFit(d, m, dispersion=disp))
    fitted.vals.each = lapply(fits, fitted.values)

    y = RS.filtered
    ybar = fitted.vals.each[[1]]

    p = c(NA, 1, 3, 7)

    disp.inv = 1 / disp
    R2s.full = sapply(2:length(fitted.vals.each), function(i) {
        mu = fitted.vals.each[[i]]
        if (length(disp) == 1 & disp[1] == 0) {
            num = rowSums(y * log(y / mu) - (y - mu), na.rm=TRUE)
            denom = rowSums(y * log(y / ybar) - (y - ybar), na.rm=TRUE)
        } else {
            num = rowSums(y * log(y / mu) - (y + disp.inv) * log((y + disp.inv) / (mu + disp.inv)), na.rm=TRUE)
            denom = rowSums(y * log(y / ybar) - (y + disp.inv) * log((y + disp.inv) / (ybar + disp.inv)), na.rm=TRUE)
        }
        (1 - (num * (n - 1)) / (denom * (n - p[i] - 1)))
    })

    adj.pseudo.R2s = t(apply(R2s.full, 1, function(r) diff(c(0, r, 1))))
    adj.pseudo.R2s = as.data.frame(adj.pseudo.R2s)
    colnames(adj.pseudo.R2s) = colnames(experimentDesign)
    adj.pseudo.R2s$Depth = rowSums(RS.filtered)
    adj.pseudo.R2s$Gene = rownames(RS.filtered)

    # remove the NA cases
    adj.pseudo.R2s = adj.pseudo.R2s[complete.cases(adj.pseudo.R2s), ]

    pseudo.R2s.m = as.data.table(melt(adj.pseudo.R2s, id=c("Gene", "Depth")))
    setnames(pseudo.R2s.m, "variable", "Level")
    pseudo.R2s.m[, Quantile:=rank(Depth) / length(Depth), by="Level"]
    pseudo.R2s.m$Method = method

    pseudo.R2s.m
}

pseudo.R2s.m = rbind(RNA.Seq.pseudoR2s(0, "Poisson Divergence"),
                     RNA.Seq.pseudoR2s(d$tagwise.dispersion, "Neg Bin Divergence"))

# modify R2s.RNA.Seq and pseudo.R2s.m to fit together
pseudo.R2s.m$Depth = NULL
levels(pseudo.R2s.m$Level) = c("Condition", "Biological", "Preparation", "Lane")

library(plyr)
geneR2s_TMM <- geneR2s %>% group_by() %>% filter(normalize.method == "TMM")

R2s.mod <- geneR2s_TMM %>% filter(technology == "RS") %>%
    dplyr::select(Gene, Level, value, Quantile) %>%
    mutate(Method = "R^2")

R2s.mod$Level = revalue(R2s.mod$Level, c("Chip/Lane"="Lane"))
R2s.combined = rbind(R2s.mod, pseudo.R2s.m)

# order them
lvls <- c("Condition", "Biological", "Preparation", "Lane")
R2s.combined <- R2s.combined %>%
    mutate(Level = factor(R2s.combined$Level, lvls)) %>%
    mutate(Method = relevel(factor(Method), "R^2"))
@

\clearpage
\bibliographystyle{unsrt}
\bibliography{refs}

<<save_image, dependson = c("merge_data", "process_R2s", "pseudo_R2", "wilcox_sets_depth")>>=
save(geneR2s, geneR2s_quantile, R2s.combined, wilcox_sets, set_pvalues, gseama_objs, file = "saved_objects.Rdata")
@

\end{document}
